### PR TITLE
sort feed by distance

### PR DESCRIPTION
**Pull request by akhrail1996**

### CODE REVIEW

File: lib/t/feeds.ex [hunk 1]
```diff
@@ -73,29 +73,45 @@ defmodule T.Feeds do
...
+    # most_liked_q(user_id, gender, gender_preferences, feeded)
+    # most_liked =
+    feed_profiles_q(user_id, gender, gender_preferences)
+    # |> maybe_apply_age_filters(min_age, max_age)
+    # |> maybe_apply_distance_filter(location, distance)
+    # |> limit(^most_liked_count)
+    # |> select_merge([p], %{distance: st_distance(^location, p.location) / 1000})
+    # select([p], %{p | distance: st_distance(^location, p.location) / 1000})
+    |> select([p], {p, fragment("st_distance(?, location) / 1000", ^location)})
+    |> order_by(fragment("st_distance(?, location) / 1000", ^location))
```
Comment (akhrail1996) [inline lib/t/feeds.ex:89]: So far I was not able to find a way to use normal order_by in Ecto, avoiding double computation (?)
related: https://stackoverflow.com/questions/58579900/order-by-virtual-computed-field

**Review by ruslandoga — COMMENTED** [review]
you can use position in `select`: 'order by 1' would order by first selected column

File: lib/t/feeds.ex (hunk 1 above)
Comment (ruslandoga) [reply lib/t/feeds.ex:89]: note that st_distance is not in meters

**Comment by ruslandoga** [comment]
also note that you shouldn't render exact distances since it allows for triangulation

**Comment by akhrail1996** [comment]
@ruslandoga yeah, do you think adding some white noise on backend is a solution?

File: lib/t/feeds.ex (hunk 1 above)
Comment (akhrail1996) [reply lib/t/feeds.ex:89]: right, thanks!

**Pull request closed without merging**